tests/integration/test_lineage.py

```diff
@@ -103,48 +116,234 @@ def test_get_column_lineage(config, ctx):
+    test_schema = test_schema_file.read_text()
+
+    uri = make_sqlalchemy_uri(scheme, username, password, at, db)
+    engine = create_engine(uri)
+    conn = engine.connect()
+
+    schema_results = conn.execute(
+        "SELECT * FROM SCHEMAS WHERE SCHEMA_NAME = 'RESERVATIONS_SCHEMA'"
+    )
+    print(schema_results)
+
+    if schema_results.returns_rows:
+        conn.execute("DROP SCHEMA RESERVATIONS_SCHEMA CASCADE")
+
+    conn.execute("CREATE SCHEMA RESERVATIONS_SCHEMA")
+    conn.execute(test_schema)
+
+    assert any(
+        "RESERVATIONS_SCHEMA" in item[0] for item in schema_results
+    ), f"{schema_results}"
+
+    # source class
+    source = HanaSource.create(config_dict, ctx)
+    source.get_db_connection()
+    inspector = inspect(conn)
+
+    view_definitions = source.get_column_lineage_view_definitions(inspector)
+
+    # sql = """SELECT
+    # H.NAME,
+    # R.TYPE,
+    # R.FREE,
+    # COUNT(R.FREE) AS FREE_RM_COUNT
+    # FROM
+    # HOTEL_SCHEMA.ROOM AS R
+    # LEFT JOIN
+    # HOTEL_SCHEMA.HOTEL AS H
+    # ON H.HNO = R.HNO
+    # GROUP BY
+    # H.NAME,
+    # R.TYPE,
+    # R.FREE;"""
+
+    for views in view_definitions:
+        # pass
+        assert views.name == "test_cross_schema"
+        assert views.schema == "reservations_schema"
+
+    lineages = list(source.get_column_view_lineage_elements(inspector))
+    assert len(lineages) == 1
+    cross_schema = lineages[0]
+    column_lineage = cross_schema[1]
+
+    # test for downstream cols
+    expected_downstreams = [
+        "name",
+        "type",
+        "free",
+        "free_rm_count",
+    ]
+    downstreams = [x[0].name for x in column_lineage]
+    assert (
+        downstreams == expected_downstreams
+    ), f"cross_schema_downstreams: {downstreams}"
+
+    # test for upstream cols
+    upstream_field_names = [
+        ["name"],  # name of hotel
+        ["type"],  # type of room
+        ["free"],  # status  of room
+        ["free"],  # status  of room
     ]
     # note that we sort the upstreams because the order is not guaranteed by sqlglot
-    upstreams = [sorted([source.name for source in x[1]]) for x in column_lineage]
+    upstreams = [sorted([source.name for source in x[1]])
+                 for x in column_lineage]
+
+    assert upstreams == upstream_field_names, f"cross_schema_upstreams: {upstreams}"
 
-    assert upstreams == upstream_field_names, f"{upstreams}"
+    # test to see the downstream shema name
+    assert (
+        column_lineage[0][0].dataset.schema == "reservations_schema"
+    ), f"cross_schema_upstreams: {column_lineage[0][0].dataset.schema}"
+
+    # test to see the source schema name
+    expected_upstream_schema = [
+        ["hotel_schema"],
+    ]
+
+    upstreams_schema = [
+        sorted([source.dataset.schema for source in x[1]]) for x in column_lineage
+    ]
+    assert upstreams_schema[0] == expected_upstream_schema[0], f"{upstreams_schema[0]}"
```

Comment:
This test seems a bit odd, shouldn't there be four values in the upstream schema list? one for each column that is referenced?